plugin/pulpcore/plugin/changeset/model.py

```diff
@@ -198,10 +295,58 @@ def save(self):
         Update the DB model to store the downloaded file.
         Then, save in the DB.
         """
-        if self.path:
-            self.model.file = File(open(self.path, mode='rb'))
-            self.model.downloaded = True
-        super(RemoteArtifact, self).save()
+        artifact = None
+
+        if not self.fetched:
+            if self._path:  # downloaded
+                try:
+                    with transaction.atomic():
```

Comment:
In each of these three cases, do you think the work being done in the ``except`` block should be done in the transaction? I've been out of the "transaction mindset" for a while now, so I'm not recalling the expected behavior.

If an ``IntegrityError`` were raised on line 307, then the model got deleted by some other connection, and then line 310 tried to fetch the model, that would be bad. Would doing it all in one transaction prevent that?